idb/postgres/postgres.go

```diff
@@ -733,10 +733,25 @@ func (db *IndexerDb) CommitRoundAccounting(updates idb.RoundUpdates, round, rewa
 			return fmt.Errorf("prepare update algo, %v", err)
 		}
 		defer setalgo.Close()
+
+		// If the account is closing the cumulative rewards field needs to be specially overwritten.
+		resetalgo, err := tx.Prepare(`INSERT INTO account (addr, microalgos, rewardsbase, rewardsTotal) VALUES ($1, $2, $3, $4) ON CONFLICT (addr) DO UPDATE SET microalgos = account.microalgos + EXCLUDED.microalgos, rewardsbase = EXCLUDED.rewardsbase, rewardsTotal = EXCLUDED.rewardsTotal`)
```

Comment:
$3 and $4 could just be 0
We can leave $2 to ensure our accounting is right.

Reply:
I'm going to fuss with this a bit in another PR which adds in createdAt/closedAt fields.

Specifically we don't need the `ON CONFLICT` bit. If an account is closing itself it must have been created already so no need for the `INSERT` part of the query.

Aside from that, I think we still need to set all of these fields in case the account was reopened in the same round that it was closed.

Reply:
I was wrong, rewardsbase should always be set to the current round, not zero.
The order of the txns should take care of the close+reopen case?
1. something to A, A gets receiver rewards, rewardsbase set to current round
2. A closes, rewardsTotal = 0, (so I still think $4 can just be 0 as applied to resetalgo)
3. something else to A, A gets no receiver rewards because rewards base is the current round, but that works the same as any other txn.

Reply:
Ohhhhkay, the complication is that a round worth of txn is being aggregated in accounting.go into an AlgoUpdate. AlgoUpdate fields are normally deltas, but sometimes we need to zero-out totalRewards instead of add a delta to it. Algo balances can still just work by deltas (and validations should prove that the Algo balance goes to zero as appropriate). Okay, I get it, this is all correct (there will never be non-zero rewards within a round where an account is closed and reopened, but the math isn't wrong to allow for that).

Reply:
I think that there 1 scenario where there would be non-zero rewards, if things work the way I think they do.

If account A is re-opened with another account closing into A, then the rewards which would have gone to the closing account would instead go to A. I didn't actually test this scenario...

Reply:
I think if another account is closed onto A, rewards to that other account are applied there first and then the balance is transferred to A. The 'Close Rewards' are what the close-to destination gets just because it's being touched.